riak/overlays/riak-cluster.sh

```diff
@@ -15,6 +16,32 @@ sed -i -r "s#distributed_cookie = (.*)#distributed_cookie = $CLUSTER_NAME#g" $RI
 sed -i -r "s#listener\.protobuf\.internal = (.*)#listener.protobuf.internal = $HOST:8087#g" $RIAK_CONF
 sed -i -r "s#listener\.http\.internal = (.*)#listener.http.internal = $HOST:8098#g" $RIAK_CONF
 
+riak_test_rapid_ready () {
+    # only make riak cluster convergence rapid if the cluster is being used for
+    # testing. otherwise, cluster communications will have a more dominant share
+    # of the overall CPU and Net I/O than they should.
+    if echo $RIAK_MODE |grep "test" >/dev/null 2>&1; then
+        # set riak_conf vnode_parallel_start, forced_ownership_handoff, and
+        # handoff_concurrency to the ring_size.
+        local ring_size=$(awk -F'=' '/ring_size/{print $2}' $RIAK_CONF |sed 's/[ ]//')
+        local handoff_concurrency=$ring_size
+        local forced_ownership_handoff=$ring_size
+        local handoff_concurrency=$ring_size
+        local sed_riak_core_section="""
+/^[ ]*{riak_core,/,/^[ ]*\]}/ {
+    s/^[ ]*{.*}[^}]*[^,]/&,/
+    s/^[ ]*\]/ \
+    {vnode_parallel_start, $vnode_parallel_start}, \
+    {forced_ownership_handoff, $forced_ownership_handoff}, \
+    {handoff_concurrency, $handoff_concurrency} \
+]/
+}
+"""
+        sed -i "$sed_riak_core_section" $RIAK_ADVANCED_CONF
+    fi
+}
+riak_test_rapid_ready
```

Comment:
I think what I'd like to do here is introduce a "proper" pre and post start mechanism and break out the chunks of code already here into their own scripts. I'm thinking about a directory called `/etc/riak/prestart.d/` and `/etc/riak/poststart.d/` where `*.sh` scripts would live. They would be sourced 1-by-1 and have access to the exported vars created here like `RIAK_CONF` et al. Most of the content of this script would be replaced by calls to source these files before and after the `start` and `wait-for-service`.